### PR TITLE
Fix CheckBox form control default value setter to use bool instead of string

### PR DESCRIPTION
**Pull request by mrle**
setting a default value for a checkbox control doesn't work correctly right now because the CheckboxFormControlBuilder.WithDefaultValue method takes a string instead of boolean.

Implementation details:

the default value specified using the WithDefaultValue method is generated as formly defaultValue: https://formly.dev/docs/examples/field-options/default-value

default value should be boolean in case of checkbox input.

We are currently generating:

`@:defaultValue: '@Html.Raw(control.DefaultValue)',`

This should be something like (in case of checkbox with default value true):

`defaultValue: true`

### CODE REVIEW

File: Enigmatry.Entry.CodeGeneration/Templates/HtmlHelperExtensions/Angular/AngularFormlyHtmlHelperExtensions.cs [hunk 1]
```diff
@@ -31,6 +31,18 @@ public static IHtmlContent GroupCssClass(this IHtmlHelper html, FormControlGroup
...
     }
 
+    public static IHtmlContent DefaultValue(this IHtmlHelper html, FormControl control)
+    {
+        if (control.DefaultValue == null)
+        {
+            return html.Raw("");
+        }
+
+        return control.FormlyType == "checkbox"
```
Comment (mrle) [inline Enigmatry.Entry.CodeGeneration/Templates/HtmlHelperExtensions/Angular/AngularFormlyHtmlHelperExtensions.cs:41]: @msljivic , not sure if we have better way to compare formlyTypes? Should I introduce a class with constants maybe, or we already have it (could find it)?

File: Enigmatry.Entry.CodeGeneration/Templates/HtmlHelperExtensions/Angular/AngularFormlyHtmlHelperExtensions.cs [hunk 2]
```diff
@@ -31,6 +31,18 @@ public static IHtmlContent GroupCssClass(this IHtmlHelper html, FormControlGroup
         return html.Raw($"fieldGroupClassName: `{classNameValue}`,\r\n");
     }
 
+    public static IHtmlContent DefaultValue(this IHtmlHelper html, FormControl control)
```
Comment (mrle) [inline Enigmatry.Entry.CodeGeneration/Templates/HtmlHelperExtensions/Angular/AngularFormlyHtmlHelperExtensions.cs:34]: Not sure if AngularFormlyHtmlHelperExtensions .cs is the right place for this extension?

File: Enigmatry.Entry.CodeGeneration.Configuration/Form/Controls/Checkbox/CheckboxFormControlBuilder.cs [hunk 3]
```diff
@@ -12,9 +13,26 @@ public CheckboxFormControlBuilder(string propertyName) : base(propertyName)
...
     {
         var checkboxFormControl = new CheckboxFormControl();
         return Build(componentInfo, checkboxFormControl);
     }
-}
\ No newline at end of file
+
+    // This method is deliberately made private so that users cannot set string default values!
+#pragma warning disable IDE0051 // Remove unused private members
+    private new CheckboxFormControlBuilder WithDefaultValue(string defaultValue)
```
Comment (mrle) [inline Enigmatry.Entry.CodeGeneration.Configuration/Form/Controls/Checkbox/CheckboxFormControlBuilder.cs:33]: @msljivic , This is the easies way to hide original implementation :) Any better ideas?

Comment (msljivic) [reply Enigmatry.Entry.CodeGeneration.Configuration/Form/Controls/Checkbox/CheckboxFormControlBuilder.cs:33]: I don't like this. We could move default value out of BaseControlBuilder / BaseControl and place it in appropriate input type e.g. string default value for input, boolean for checkbox etc.

The other option is to make API similar to HTMLInputElement, which has both defaultValue: string and defaultChecked: boolean properties.

I would try the first option.

File: Enigmatry.Entry.CodeGeneration.Configuration/Form/Controls/FormControl.cs [hunk 4]
```diff
@@ -49,4 +48,6 @@ public virtual void ApplyValidationConfiguration(IEnumerable<IFormlyValidationRu
     }
 
     public bool IsRequired => ValidationRules.HasRule<IsRequiredValidationRule>();
-}
\ No newline at end of file
+
+    public abstract string? DefaultValueAsString();
```
Comment (msljivic) [inline Enigmatry.Entry.CodeGeneration.Configuration/Form/Controls/FormControl.cs:52]: I would remove this completely. 
Then, in the html helper switch on types that have default value (Input, Checkbox, etc.)

**Review by msljivic — APPROVED** [review]

**Pull request merged**